### DISCUSSION TITLE
Possible bug? json metadata naming with z-layers

### DISCUSSION THREAD

**Discussion by miaouxtoo** [nftchef/art-engine#157] [Help + Q&A]
Hi guys, 

I've found that if you use nested directories AND an overwrite with z-layers in a directory, this forces the trait name to default to the parent folder. 

E.g. we have: 

HairTraitTest#10/
-z3,ear-on-top.png
-hair-A-fixme#10.png
-hair-B-fixme#10.png

When this is used in generation, the json trait name becomes: "HairTraitTest" instead of either hair-A-fixme or hair-B-fixme as it would normally (if z3,ear-on-top.png) didn't exist. This obfuscates the actual distinct json trait that's used.

Not sure if this is intentional... or if it's just me. 
The kludge workaround is to make 2 subfolders, one for hair-A-fixme#10/ (include the z3 and hair-A file), and another for hair-B-fixme#10/ (include the z3 and hair-B file). 

The only issue with the workaround is that although the trait names are correct, there is also the issue that the immediate parent folder is extra... So now rarity is labeled "Hair -> HairTraitTest -> hair-A-fixme" , rather than the tidier: "Hair -> hair-A-fixme" expected.

Thanks, 

Great work NFTChef

**Comment by BakeSloth** [comment]
I worked around this by nesting my folders.
Make sure all files were nested 2 folder deep. (or more if you need)
root/layer/folder1/folder2/file.png

then edited line 202 on main.js by changing typeAncestor to 2
keeping useRootTraitType = true; on config.js
this gives you more control over naming your root folder
